app/controllers/administrate/application_controller.rb

```diff
@@ -140,7 +140,15 @@ def find_resource(param)
     end
 
     def scoped_resource
-      resource_class.default_scoped
+      resource_class_for_current_action.default_scoped
+    end
+
+    def resource_class_for_current_action
+      resource_class_for_action(action_name)
+    end
+
+    def resource_class_for_action(_action_name)
+      resource_class
     end
```

Comment:
Hi Pablo!
I'm surprised to see this method simply responds with `resource_class`, and seemingly does not depend on `_action_name`.

Is your idea that a developer using our library could subclass and override `resource_class_for_action(...)`, thereby making use of `_action_name`?

If so, the approach seems good. I hope I'm reading your code properly.

Reply:
That's right, as you find out later below :-)